src/pages/en/reference/directives-reference.md

```diff
@@ -0,0 +1,195 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Directives Reference
+---
+
+Directives are special properties that can be passed to HTML elements, Astro components, And UI Framework components. Directives can only be used inside of Astro components.
+
+The directives on this page are originised by what they can be used on.   
+
+## UI Framework components
+
+These directives can be used on components from any of the UI Frameworks that Astro supports. Lean more about UI Framework components on their [dedicated page](/en/core-concepts/framework-components).
+
+### `client:load`
+
+Start importing the component JS at page load.
+
+💡 *Useful for immediately-visible UI elements that need to be interactive as soon as possible.*
+
+### `client:idle`
+
+Start importing the component JS as soon as main thread is free.
+
+💡 *Useful for items that don't need to be immediately interactive.*
+
+### `client:visible`
+
+Start importing the component JS as soon as the element enters the viewport.
+
+💡 *Useful for content lower down on the page.*
+
+### `client:media={QUERY}`
+
+Start importing the component JS as soon as the browser matches the given media query.
+
+💡 *Useful for sidebar toggles, or other elements that might be used only for certain screen sizes.*
+
+> ⚠️ Remember, this directive only refers to making the component *interactive* at certain media queries. This does not affect the component being *rendered to the screen*, nor its *visibility*!
+
+### `client:only=" "`
+
+Start importing the component JS at page load, similar to `client:load`.
+
+💡 *Useful for components that are entirely dependent on client-side APIs.* 
+
+>⚠️ This component will be **skipped** at build time, and to assist the client, you should specify which renderer to use from the array in your [`astro.config.mjs` configuration](/en/reference/configuration-reference).
+>
+> e.g. `<client:only="react" />` or `<client:only="my-custom-renderer" />`
+
+## Script and Style tags
+
+These directives can be used on HTML `<script>` and `<style>` tags.
```

Comment:
(NB: I originally edited this to say "used on both" script and style tags, but then when I read further, I realized that this entire section wasn't just things that can be used on both. That's what led to my long comment above.)